comit/src/lib.rs

```diff
@@ -27,6 +27,13 @@ pub use self::{
 };
 use digest::ToDigestInput;
 
+#[derive(Clone, Copy, Debug)]
+pub struct Swap {
```

Comment:
I would refrain from adding things like these to the comit lib for now. If it turns out that we really need something like these in all the places, then we can always add it. But now is too early IMO.

Reply:
This is actually what got me out of bed to hack on this now, I've moved this new type to the db module. It is foreseeable that the `http_api::Swap` could be replaced now with this swap since this is arguably the top level view of a swap from the databases perspective and since this type is used solely to get the role and swap protocol instance (i.e. `Herc20Halight).

Reply:
Happy to add it at the DB level.